docs/hugo/content/contributing/generator-overview.md

```diff
@@ -0,0 +1,56 @@
+
+The key packages used to structure the code of the generator are as follows:
+
+| Package          | Content                                                                                                                                               |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `astmodel`         | Short for _Abstract Syntax Tree Model_, this package contains the core data types for defining the Go functions, data types, and methods we generate. |
+| `functions`        | Support for generation of individual functions, based on the `astmodel.Functions` interface                                                           |
+| `interfaces`       | Support for generation of interface implementations, based on the `astmodel.InterfaceImplementer` interface                                           |
+| `testcases`        | Support for generation of test cases (used to verify our generated code works as expected), based on the `astmodel.TestCase` interface                |
+| `astbuilder`       | Intention revealing utility methods for creating the underlying Go abstract syntax tree we serialize as the last step of generation.                  |
+| `codegen`          | The core processing pipeline of the code generator                                                                                                    |
+| `codegen/pipeline` | Individual pipeline stages that are composed to form the code generator itself.                                                                       |
+| `test`             | Support methods to make writing tests easier                                                                                                          |
+
+## Object Model
+
+At the core, the code generator works with a rich semantic object model that captures the structure of the resources (and related types) we are generating.
+
+Underpinning this object model is the `astmodel` package.
+
+The interface `astmodel.Type` represents a specific data type. The most widely used implementations of `Type` fall into a few separate groups:
+
+* Simple Go types, including `PrimitiveType`, `ArrayType`, and `MapType`.
+* Complex types with internal structure, most notably `ObjectType` and `ResourceType`.
+* Wrapper types that provide additional semantics, including `OptionalType`, `ErroredType`, `FlaggedType` and `ValidatedType`.
+
+This list is not exhaustive; other implementations of `Type` are used within limited scopes. For example, `AllOfType` and `OneOfType` represent JSON Schema techniques for creating object definitions via composition.
+
+Usefully, there is also `TypeName` which is both a type in itself and an indirect reference to a type defined elsewhere.
+
+When a `Type` is given a `TypeName`, it becomes a `TypeDefinition` and can be emitted as the source code for a Go type definition.
```

Comment:
Might be useful to add something like - `TypeName` consists of `Name` and `PackageRef` for the `Type`

Reply:
I'm not sure - want to keep this at a high level and more as a guide to "here are useful things to look at"; if it starts including too much detail it stops being an overview.